examples/opensora_pku/opensora/dataset/loader.py

```diff
@@ -1,91 +1,234 @@
+# Copyright 2023 Huawei Technologies Co., Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+import gc
+import logging
+import random
+from collections import defaultdict
+
+import numpy as np
+
 import mindspore as ms
+from mindspore.communication.management import get_local_rank, get_local_rank_size
+from mindspore.dataset import GeneratorDataset
 
-from .image_dataset import ImageDataset
-from .video_dataset import VideoDataset
+_logger = logging.getLogger(__name__)
 
 
 def create_dataloader(
-    ds_config,
+    dataset,
     batch_size,
-    ds_name="image",
+    column_names=["video"],
     num_parallel_workers=12,
-    max_rowsize=32,
+    max_rowsize=64,
```

Comment:
max_rowsize too large may lead to CPU momory OOM

Reply:
OK, changed to 32.